WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -149,16 +151,12 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
         NotificationCenter.default.addObserver(self, selector: #selector(showAddSelfHostedSite), name: .addSelfHosted, object: nil)
     }
 
-    private func setupSegmentedControl() {
-        segmentedControlContainerView.isHidden = !FeatureFlag.mySiteDashboard.enabled
-
-        segmentedControl.removeAllSegments()
-        Section.allCases.forEach { section in
-            segmentedControl.insertSegment(withTitle: section.title, at: section.rawValue, animated: false)
+    private func setupSegmentedControl(for blog: Blog) {
+        guard FeatureFlag.mySiteDashboard.enabled else {
+            return
         }
-        segmentedControl.selectedSegmentIndex = 0
+        segmentedControlContainerView.isHidden = !blog.isHostedAtWPcom || UIDevice.isPad()
```

Comment:
`blog.isHostedAtWPcom` doesn't take into account Jetpack connected sites (including Atomic), and the dashboard is supported for them.

Reply:
Fixed! 46a0d73

I added these use cases to the testing instructions above, as well as a video.